### PR TITLE
Allow throwing errors across async boundaries

### PR DESCRIPTION
**Pull request by benweatherman**
Throw errors to simplify the code slightly

### Other detritus
- Use a tsconfig for node14 cause that's the max build target supported by netlify (neé aws) https://docs.netlify.com/functions/optional-configuration/?fn-language=ts#node-js-version-for-runtime. I continued to use node16 in the package file since netlify will use that for building everything.
- Set `autoLaunch = false` so the netlify dev server doesn't try to open a (broken) page when it starts

### CODE REVIEW

**Comment by EverlastingBugstopper** [comment]
LGTM!

**Review by EverlastingBugstopper — APPROVED** [review]

**Comment by EverlastingBugstopper** [comment]
i'm... not quite sure why it's complaining but it says "this branch cannot be rebased due to conflicts"... - if i run `git merge avery/tarball-endpoint` from the CLI it says `Already up to date.` so i'm very confused

File: src/lib/binary.ts [hunk 1]
```diff
@@ -54,46 +53,37 @@ export class Binary {
...
+    let response = await fetch(versionUrl, {
+      method: "HEAD",
+      redirect: "manual",
+    });
+    if (response?.status === 301 || response?.status === 302) {
+      let realLatestUrl = response.headers.get("location");
+      // https:, , github.com, apollographql, rover, releases, v0.99.99
+      const urlComponents = realLatestUrl?.split("/");
+      // grab the last element
+      const latestVersion = urlComponents?.pop();
```
Comment (EverlastingBugstopper) [inline src/lib/binary.ts:67]: nice lol

File: package.json [hunk 2]
```diff
@@ -25,8 +25,8 @@
     "prettier": "^2.5.1"
   },
   "dependencies": {
-    "@babel/plugin-transform-runtime": "^7.17.0",
```
[the remark this answers is not in the document]
Comment (benweatherman) [reply package.json:28]: i think this was just npm reordering things

**Pull request merged**